packages/libs/deer/README.md

```diff
@@ -0,0 +1,113 @@
+
+`deer` solves this problem, by returning every issue present. This means that a single API call with the payload given will result in the errors: `256 larger than u8::MAX`, `null is not String`, and `extra key "extra" provided`.
+
+This in turn also means that deer can be used for custom validation while deserializing, while still being able to return all validation issues.
+
+<sub>
+deer might provide a way in the future to describe these constraints.
+</sub>
+
+## Limitations
+
+deer currently does **not** parse values itself, but relies on external parsers like `serde_json`, this means that parsing will be fail-fast, and deer only touches syntactically correct values.
+
+## Future Plans
+
+The first release of `deer` is intentionally minimal and tries to lay a good foundation to extend functionality in the future. There are many future possible directions and ideas we're trying to see if they can benefit the different use cases.
+
+### Introspection Support
+
+Currently, popular crates like `serde` do not provide a way to introspect what the output will be. Other tools try to fill the gap by manually interpreting the instructions given to these crates. This often leads to edge cases, resulting in the dissonance between the expected value and reality. The goal with the explicit support of introspection is to allow other tools to make use of it and build abstractions around it instead of trying to reverse engineer.
+
+<sup>
+How the introspected format might look like is currently unknown.
+</sup>
+
+### Validation
+
+Currently, to be able to do validation, one must create a new type that performs the validation step. This extra boilerplate is often a pain point. The idea is to instead allow for _optional_ validation via combinators using the derive macro.
+
+<details>
+<summary>What it may look like</summary>
+
+```rust
+#[derive(deer::Deserialize)]
+struct Payload {
+    #[validate(all(min(12), max(24)))]
+    length: u8
+}
+```
+
+</details>
+
+### Lax Deserialization
+
+Instead of strictly deserializing types, one might prefer to deserialize while coercing values (`"1"` might be interpreted as `1` instead). This behavior would be opt-in instead of opt-out and enabled on the `derive` level.
+
+### Deserializer with Parser
+
+`deer` currently relies on external tools (notably `serde`) to implement parsing of different formats, which means that we still fail-fast during the parsing of malformed input. In the future, `deer` might provide its own parser, which tries to recover from parsing errors and still provide parsing diagnostics.
+
+<details>
+<summary>What it may look like</summary>
+
+```json
+{
+  "i8": "string"
+```
```

Comment:
Would us picking something other than JSON avoid this? 
<img width="788" alt="image" src="https://user-images.githubusercontent.com/25749103/194579717-8301d1dc-95cb-4d3d-b29c-facede10a27e.png">

Because I think we're going to have to deal with this on every PR for this text, not just this one.

Reply:
I removed formatting for now as I wasn't able to find a way to display incomplete JSON otherwise ([99538ec](https://github.com/hashintel/hash/pull/1163/commits/99538ec20ea56c29dc8876b1ebfc8fc61cf204c0))

Reply:
Dang that's quite annoying. Perhaps we can revisit later on